src/marshmallow/fields.py

```diff
@@ -1243,6 +1243,9 @@ def _serialize(self, value: _D | None, attr, obj, **kwargs) -> str | float | Non
         return value.strftime(data_format)
 
     def _deserialize(self, value, attr, data, **kwargs) -> _D:
+        internal_type: type[_D] = getattr(dt, self.OBJ_TYPE)
```

Comment:
could also add a separate class attribute, like `INTERNAL_TYPE`. i don't have a strong feeling either way

Reply:
The perf impact must be minimal. We can keep this.